fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/predicates/PredicateWithValueAndRanges.java

```diff
@@ -234,6 +235,14 @@ public static PredicateWithValueAndRanges ofRanges(@Nonnull final Value value, @
         return new PredicateWithValueAndRanges(value, ranges);
     }
 
+    @Nonnull
+    @Override
+    public PredicateWithValueAndRanges translateValue(@Nonnull final UnaryOperator<Value> translator) {
+        final var newValue = Verify.verifyNotNull(this.getValue().replace(translator));
```

Comment:
Can you change this to just:

```
Verify.verifyNotNull(translator.apply(this.getValue());
```

This is more powerful as you can put the `replace()` call into the lambda you hand in and you can truly just swizzle the old `Value` tree for a newly constructed one. It's also the way this is done in `Comparison`.

Can you do the same for all the other implementations?